src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,116 @@
+        self._dir = tempfile.TemporaryDirectory(prefix="test-chroot-")
+
+        paths: typing.Dict[str, str] = {
+            "insights_client.MOTD_SRC": "/etc/insights-client/insights-client.motd",
+            "insights_client.MOTD_FILE": "/etc/motd.d/insights-client",
+            "insights_client.REGISTERED_FILE": "/etc/insights-client/.registered",
+            "insights_client.UNREGISTERED_FILE": "/etc/insights-client/.unregistered",
+        }
+
+        self.patches: typing.List[unittest.mock.patch] = []
+        for target, path in paths.items():
+            mocked_path = f"{self.chroot}{path}"
+            pathlib.Path(mocked_path).parent.mkdir(parents=True, exist_ok=True)
+            patch = unittest.mock.patch(target, mocked_path)
+            self.patches.append(patch)
+            patch.start()
+
+        with (self.chroot / "etc/insights-client/insights-client.motd").open("w") as f:
+            f.write("This is the content of the MOTD file.")
+
+    @property
+    def chroot(self) -> pathlib.Path:
+        return pathlib.Path(self._dir.name)
+
+    def __del__(self):
+        for patch in self.patches:
+            patch.stop()
+        self._dir.cleanup()
+
+
+@pytest.fixture(scope="function")
+def mock_fs():
+    fs = MockFS()
+    yield fs
+    del fs
+
+
+def test_present(mock_fs):
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+    # The file gets created/symlinked when .registered & .unregistered do not exist
+    insights_client.update_motd_message()
+    assert (mock_fs.chroot / "etc/insights-client/insights-client.motd").samefile(
+        mock_fs.chroot / "etc/motd.d/insights-client"
+    )
+    assert (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+
+def test_absent_on_dot_registered(mock_fs):
+    # The MOTD file exists by default...
+    insights_client.update_motd_message()
+    assert (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+    # ...and gets removed when .registered exists.
+    (mock_fs.chroot / "etc/insights-client/.registered").open("w").close()
+
+    insights_client.update_motd_message()
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+    # It stays absent when run multiple times.
+    insights_client.update_motd_message()
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+
+def test_absent_on_dot_unregistered(mock_fs):
+    # The MOTD file exists by default...
+    insights_client.update_motd_message()
+    assert (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+    # ...and gets removed when .unregistered exists.
+    (mock_fs.chroot / "etc/insights-client/.unregistered").open("w").close()
+
+    insights_client.update_motd_message()
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
+
+    # It stays absent when run multiple times.
+    insights_client.update_motd_message()
+    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
```

Comment:
This [too](https://github.com/RedHatInsights/insights-client/pull/202/files/d5a2300b8eda44d6607892307a67107ad5dfc14b#diff-cd2e501f6e57473e8f3436595c4f2eca1ca8ee2d50c9bac72b228fa219d42012) is rather a separate use case. Up to here, the test verifies that the MOTD file gets removed if _.uregistered_ file is present. That is remains absent (doesn’t get created) if it already doesn’t exist is a different scenario.

```suggestion

def test_still_absent_on_dot_unregistered(mock_fs):
    # It stays absent when run multiple times.
    (mock_fs.chroot / "etc/insights-client/.unregistered").open("w").close()
    insights_client.update_motd_message()
    assert not (mock_fs.chroot / "etc/motd.d/insights-client").exists()
```

Reply:
The test verifies that it is absent if other file exists. It is still one test and one use-case, only two paths to get there.

Reply:
Which makes it worth separate test functions.